solutions/usaco-688.mdx

```diff
@@ -0,0 +1,60 @@
+---
+id: usaco-688
+source: USACO Bronze 2017 January
+title: Hoof, Paper, Scissors
+author: Ryan Chou
+---
+
+## Solution 1
+[Official Analysis](http://www.usaco.org/current/data/sol_hps_bronze_jan17.html)
+
+## Solution 2 - Generating All Permutations
+
+We can map the games to every possible permutation of the moves (hoof, paper, scissors) and find the sequence which gives us the maximum wins. This works because $1 \leq N \leq 100$.
+
+### Implementation
+
+**Time Complexity: ** $\mathcal{O}(n)$.
+<LanguageSection>
+<PySection> 
+
+```py
+from itertools import permutations
+import sys
+
+sys.stdin = open('hps.in', 'r')
+sys.stdout = open('hps.out', 'w')
+
+n = int(input())
+games = []
+
+for _ in range(n):
+	games.append(list(map(int, input().split())))
+
+# Original Ordering 
+# 1 = Hoof, 2 = Paper, 3 = Scissors
+choices = [1, 2, 3]
+
+# in these cases, cow 1 wins
+# hoof beats scissors, 
+# scissors beats paper, 
+# and paper beats hoof
+scenarios = [[1, 3], [3, 2], [2, 1]]
+
+# generates all permutations
+pers = list(permutations(choices, 3))
+
+maxwins = 0
+
```

Comment:
46 and 48 seem like unnecessary whitespace

Reply:
> some explanation in the for loop would be nice? but maybe i'm just dumb

thanks for the suggestion, adding it now. :)